util/src/main/java/io/kubernetes/client/util/KubeConfig.java

```diff
@@ -61,22 +68,35 @@ public KubeConfig(ArrayList<Object> contexts,
         this.users = users;
     }
 
-    public void setContext(String context) {
+    public boolean setContext(String context) {
         currentCluster = null;
         currentUser = null;
-        currentContext = (Map<String, Object>) findObject(contexts, context).get("context");
+        Map<String, Object> ctx = findObject(contexts, context);
+        if (ctx == null) {
+            System.err.println("Failed to find context: " + context + " in " + contexts);
```

Comment:
The stderr thing distracted me. I agree that returning false is better here. Can you get rid of the stderr logging?